ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +71,50 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
-    top: Math.floor(heightWithMargins * pageIndex + marginTopPixels + (height - bottomPixels)),
+  const pageId = generatePageIdFromIndex(pageIndex);
+
+  const pageIdElement = document.getElementById(pageId);
+
+  if (!pageIdElement) {
+    return;
+  }
+
+  const parentElement = getScrollParent(pageIdElement);
+
+  if (!parentElement) {
+    return;
+  }
+
+  parentElement.scrollTo({
```

Comment:
These newlines are kinda superfluous. You could easily remove them and keep the code readable.

```suggestion
  // Find page element
  const pageId = generatePageIdFromIndex(pageIndex);
  const pageIdElement = document.getElementById(pageId);
  if (!pageIdElement) {
    return;
  }

  // Find first scrollable parent
  const parentElement = getScrollParent(pageIdElement);
  if (!parentElement) {
    return;
  }

  // Apply scroll
  parentElement.scrollTo({
```